clang/test/SemaCXX/attr-lifetimebound.cpp

```diff
@@ -47,6 +47,29 @@ namespace usage_ok {
     q = A(); // expected-warning {{object backing the pointer q will be destroyed at the end of the full-expression}}
     r = A(1); // expected-warning {{object backing the pointer r will be destroyed at the end of the full-expression}}
   }
+
+  struct FieldCheck {
+    struct Set {
+      int a;
+    };
+    struct Pair {
+      const int& a;
+      int b;
+      Set c;
```

Comment:
Can you add a test case of pointer type file, e.g. `int* d;` to ensure that we will not fire a diagnostic on `const int* d = FieldCheck{x}.getR().d;`?

Reply:
Done.